plutus-extra/src/Plutus/Contract/Test/Extra.hs

```diff
@@ -267,57 +294,30 @@ dataAtComputedAddressWithState ::
   (w -> datum -> Bool) ->
   TracePredicate
 dataAtComputedAddressWithState contract inst addressGetter check =
-  utxoAtComputedAddressWithStateImpl contract inst addressGetter $ \w addr utxoMap -> do
+  utxoAtComputedAddress contract inst addressGetter $ \w addr utxoMap ->
     let datums = mapMaybe (uncurry $ getTxOutDatum @datum) $ Map.toList utxoMap
-        result = any (check w) datums :: Bool
-    unless result $
-      tell @(Doc Void)
-        ( "Data at address" <+> pretty addr <+> "was"
-            <+> foldMap (foldMap pretty . Ledger.txData . Ledger.txOutTxTx) utxoMap
-            <> line
-            <> "Contract writer data was"
-            <+> pretty w
-        )
-    return result
-
-{- | Extract UTxOs at a computed address and call continuation returning
- Boolean value based on both the address and UTxOs.
- The address is computed using data acquired from contract's writer instance.
-
-  @since 1.1
--}
-utxoAtComputedAddress ::
-  forall
-    (effs :: [Type -> Type])
-    (w :: Type)
-    (s :: Row Type)
-    (e :: Type)
-    (a :: Type)
-    (contract :: Type -> Row Type -> Type -> Type -> Type).
-  ( Member (Error Folds.EmulatorFoldErr) effs
-  , Member (Writer (Doc Void)) effs
-  , Monoid w
-  , IsContract contract
-  ) =>
-  -- | The 'IsContract' code
-  contract w s e a ->
-  -- | The 'ContractInstanceTag', acquired inside the
-  -- 'Plutus.Trace.Emulator.EmulatorTrace'
-  ContractInstanceTag ->
-  -- | The function computing 'Ledger.Address'
-  (w -> Maybe Ledger.Address) ->
-  -- | The continuation function acting as a predicate
-  (Ledger.Address -> UtxoMap -> Eff effs Bool) ->
-  Folds.EmulatorEventFoldM effs Bool
-utxoAtComputedAddress contract inst addressGetter cont =
-  utxoAtComputedAddressWithStateImpl contract inst addressGetter (const cont)
+     in showStateIfFailAndReturn
+          [CheckedDatas utxoMap, CheckedWriter w]
+          addr
+          (any (check w) datums)
 
 {- | Check that the UTxO at a computed address
  and data aquired from contract's writer instance meet some condition.
  The address is computed using data acquired from contract's writer instance.
 
   @since 1.1
 -}
+showStateIfFailAndReturn ::
```

Comment:
Did the Haddock above become detached from the function it was supposed to document?

Reply:
Oh right, it was :sweat_smile:
Gonna fix that.